app/model/repository/AssignmentSolutions.php

```diff
@@ -86,44 +87,90 @@ private function findValidSolutionsForAssignments(array $assignments, User $user
   }
 
   /**
-   * Find best solutions of given assignments for user.
+   * Local function used to determine the best solution in a collection.
+   * @param AssignmentSolution|null $best The best solution found so far
+   * @param AssignmentSolution $solution Solution to be compared with the best solution
+   * @return AssignmentSolution Better of the two given solutions
+   */
+  private static function compareBestSolution(?AssignmentSolution $best, AssignmentSolution $solution)
+  {
+    if ($best === null) {
+      return $solution;
+    }
+
+    if ($best->isAccepted()) {
+      return $best;
+    }
+
+    if ($solution->isAccepted()) {
+      return $solution;
+    }
+
+    if ($best->getTotalPoints() < $solution->getTotalPoints()
+      || ($best->getTotalPoints() === $solution->getTotalPoints()
+      && $best->getSolution()->getCreatedAt() < $solution->getSolution()->getCreatedAt())) {
```

Comment:
This line is unnecessary, solutions are ordered in database, either remove this line, or ordering in database...